blockchain.go

```diff
@@ -449,8 +459,18 @@ func NewBlockchain(opts ...Option) (*Blockchain, error) {
 	if err != nil {
 		return nil, err
 	}
-	return b, nil
+	if conf.WithContracts {
+		deployments, err := DeployContracts(b)
+		if err != nil {
+			conf.ServerLogger.Error().Err(err).Msg("❗  Failed to deploy contracts")
+		}
 
+		for _, contract := range deployments {
+			conf.ServerLogger.Info().Fields(map[string]any{
```

Comment:
Two side effects of the logging taking place inside the `Blockchain`, are the following:

```bash
  flow-code-coverage git:(main) ✗ ~/Dev/forks/flow-cli/cmd/flow/flow-x86_64-linux- emulator --coverage-reporting --contracts       
5:30PM INF ⚙️   Using service account 0xf8d6e0586b0a20c7 serviceAddress=f8d6e0586b0a20c7 serviceHashAlgo=SHA3_256 servicePrivKey=2619878f0e2ff438d17835c2a4561cb87b4d24d72d12ec34569acd0dd4af7c21 servicePubKey=ebd3b08b2926ae27605f6ab060af8e0efd53f638a2664c55bdc76080ae0d2ad42b3a2835dce6059a54702df26296b24b70c575fa69f965228542dc4349fa6c04 serviceSigAlgo=ECDSA_P256
5:30PM INF 💵  FUSD contract                             FUSD=0xf8d6e0586b0a20c7
5:30PM INF ✨  NFT contract                              NonFungibleToken=0xf8d6e0586b0a20c7
5:30PM INF ✨  Metadata views contract                   MetadataViews=0xf8d6e0586b0a20c7
5:30PM INF ✨  Example NFT contract                      ExampleNFT=0xf8d6e0586b0a20c7
5:30PM INF ✨  NFT Storefront contract v2                NFTStorefrontV2=0xf8d6e0586b0a20c7
5:30PM INF ✨  NFT Storefront contract                   NFTStorefront=0xf8d6e0586b0a20c7
5:30PM INF 📜  Flow contract                             FlowStorageFees=0xf8d6e0586b0a20c7
5:30PM INF 📜  Flow contract                             FlowServiceAccount=0xf8d6e0586b0a20c7
5:30PM INF 📜  Flow contract                             FlowToken=0x0ae53cb6e3f42a79
5:30PM INF 📜  Flow contract                             FungibleToken=0xee82856bf20e2aa6
5:30PM INF 📜  Flow contract                             FlowFees=0xe5a8b7f23e8b548f
```
The common contracts are printed before the system contracts.

And second, the deployment info logs, are printed in the `test` command:

```bash
➜  flow-code-coverage git:(main) ✗ ~/Dev/forks/flow-cli/cmd/flow/flow-x86_64-linux- test --cover test_string_utils.cdc
Running tests...⠇
5:31PM INF 💵  FUSD contract FUSD=0xf8d6e0586b0a20c7
5:31PM INF ✨  NFT contract NonFungibleToken=0xf8d6e0586b0a20c7
5:31PM INF ✨  Metadata views contract MetadataViews=0xf8d6e0586b0a20c7
5:31PM INF ✨  Example NFT contract ExampleNFT=0xf8d6e0586b0a20c7
5:31PM INF ✨  NFT Storefront contract v2 NFTStorefrontV2=0xf8d6e0586b0a20c7

Test results: "test_string_utils.cdc"
- PASS: testFormat
- PASS: testExplode
- PASS: testTrimLeft
- PASS: testTrim
- PASS: testReplaceAll
- PASS: testHasPrefix
- PASS: testHasSuffix
- PASS: testIndex
- PASS: testCount
- PASS: testContains
- PASS: testSubstringUntil
- PASS: testSplit
- PASS: testJoin
Coverage: 84.4% of statements
```

I do not quite like that :sweat_smile:

Reply:
Update: This is all resolved after the refactoring suggested by @bluesign